### PR TITLE
Swap Improvements

### PR DESCRIPTION
**Pull request by dmosites**

### CODE REVIEW

**Review by Perelyn-sama — COMMENTED** [review]
Hey Don, I don't get why you changed the second argument of `safeTransferFrom` on line 227(Swap.sol) from  `msg.sender` to `recipent`. Since the recipient is expected to be the null address for public swaps and the plan to substitute `msg.sender` for it.

File: source/swap/contracts/Swap.sol [hunk 1]
```diff
@@ -8,6 +8,8 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "./interfaces/ISwap.sol";
 
+import "hardhat/console.sol";
```
Comment (dmosites) [inline source/swap/contracts/Swap.sol:11]: Please remove this and the related lines below @Perelyn-sama

**Comment by dmosites** [comment]
> Hey Don, I don't get why you changed the second argument of `safeTransferFrom` on line 227(Swap.sol) from `msg.sender` to `recipent`. Since the recipient is expected to be the null address for public swaps and the plan to substitute `msg.sender` for it.

The aim is for the sender to be able to transfer their proceeds to an arbitrary `recipient`. In most cases the sender will just use their own wallet address for the `recipient` argument.

File: source/swap/contracts/Swap.sol (hunk 1 above)
Comment (Perelyn-sama) [reply source/swap/contracts/Swap.sol:11]: On it, sorry about that.

**Comment by Perelyn-sama** [comment]
> > Hey Don, I don't get why you changed the second argument of `safeTransferFrom` on line 227(Swap.sol) from `msg.sender` to `recipent`. Since the recipient is expected to be the null address for public swaps and the plan to substitute `msg.sender` for it.
> 
> The aim is for the sender to be able to transfer their proceeds to an arbitrary `recipient`. In most cases the sender will just use their own wallet address for the `recipient` argument.

Okay, I guess that makes sense. Thanks for clarifying.

File: source/swap/contracts/Swap.sol [hunk 2]
```diff
@@ -787,13 +693,15 @@ contract Swap is ISwap, Ownable {
...
-      msg.sender,
+      senderWallet,
       senderToken,
       senderAmount
     );
 
     // Recover the signatory from the hash and signature
     address signatory = _getSignatory(hashed, v, r, s);
+    // console.log(signatory);
+    // console.log(signerWallet);
```
Comment (dmosites) [inline source/swap/contracts/Swap.sol:704]: Sorry should have flagged these as well @Perelyn-sama. Thanks.

**Pull request merged**